pkg/printers/humanreadable.go

```diff
@@ -195,6 +268,42 @@ func (h *HumanReadablePrinter) PrintObj(obj runtime.Object, output io.Writer) er
 			h.printHeader(headers, w)
 			h.lastType = t
 		}
+
+		if !handler.rowPrintFunc.IsNil() {
+			args := []reflect.Value{reflect.ValueOf(obj), reflect.ValueOf(h.options)}
```

Comment:
I agree.  On the other hand, I am much more concerned with working correctly in the current code path than completely rewriting printers from scratch now.